app.py

```diff
@@ -86,6 +133,22 @@ def handleJoin(self, room, message):
             self.sendPm(user.id, self.usernotes.pendingMessages(user.id))
 
     def parseMessage(self, msg, roomName):
+        """Parses the message given by a user and delegates the tasks further
+
+        This is where we handle most of the parsing of the PS protocols. Tasks
```

Comment:
Only non-battle related protocol is handled here. Battle protocol has its own dedicated handler. `splitMessage` will judge based on what room the message was sent from what should be done with it, and which handler should be used to parse the message.